mantle/cmd/kola/testiso.go

```diff
@@ -154,37 +196,38 @@ func runTestIso(cmd *cobra.Command, args []string) error {
 			}
 			fmt.Printf("Successfully tested legacy installer for %s\n", kola.CosaBuild.Meta.OstreeVersion)
 		}
-	} else if legacy {
+	} else if _, ok := targetScenarios[scenarioLegacyInstall]; ok {
 		return fmt.Errorf("build %s has no legacy installer kernel", kola.CosaBuild.Meta.Name)
 	}
 
 	foundLive := kola.CosaBuild.Meta.BuildArtifacts.LiveKernel != nil
-	if !nolive {
+	if _, ok := targetScenarios[scenarioPXEInstall]; ok {
 		if !foundLive {
 			return fmt.Errorf("build %s has no live installer kernel", kola.CosaBuild.Meta.Name)
 		}
-		if !nopxe {
-			ranTest = true
-			instPxe := baseInst // Pretend this is Rust and I wrote .copy()
 
-			if err := testPXE(instPxe); err != nil {
-				return err
-			}
-			printSuccess("PXE")
-		}
+		ranTest = true
+		instPxe := baseInst // Pretend this is Rust and I wrote .copy()
 
-		if !noiso {
-			ranTest = true
-			instIso := baseInst // Pretend this is Rust and I wrote .copy()
-			if err := testLiveIso(instIso, completionfile); err != nil {
-				return err
-			}
-			printSuccess("ISO")
+		if err := testPXE(instPxe); err != nil {
+			return err
+		}
+		printSuccess("PXE")
+	}
+	if _, ok := targetScenarios[scenarioISOInstall]; ok {
+		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
+			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
+		}
+		ranTest = true
+		instIso := baseInst // Pretend this is Rust and I wrote .copy()
+		if err := testLiveIso(instIso, completionfile); err != nil {
+			return err
 		}
+		printSuccess("ISO")
 	}
 
 	if !ranTest {
-		return fmt.Errorf("Nothing to test!")
+		panic("Nothing was tested!")
```

Comment:
Quick question, is there a specific reason why use `panic` instead of returning error?

Reply:
We should now only hit this if the code above is buggy:
```
	if len(targetScenarios) == 0 {
		return fmt.Errorf("No scenarios specified!")
	}
```

Reply:
:+1: I see..